kotlin/setup-location-driven-geotriggers/src/main/java/com/esri/arcgisruntime/sample/setuplocationdrivengeotriggers/MainActivity.kt

```diff
@@ -0,0 +1,380 @@
+            // If the user exits a given geofence, remove the feature's information from the UI
+            removeFeatureInformation(
+                fenceFeatureName,
+                geotriggerNotificationInfo.geotriggerMonitor.geotrigger.name
+            )
+        }
+    }
+
+    /**
+     * Creates a [GardenSection] using [fenceGeotriggerNotificationInfo] and
+     * adds the [fenceFeatureName] to the HashMap of visited sections
+     */
+    private fun addFeatureInformation(
+        fenceFeatureName: String,
+        fenceGeotriggerNotificationInfo: FenceGeotriggerNotificationInfo
+    ) {
+
+        // Recenter the camera on the user if need be
+        mapView.locationDisplay.autoPanMode = LocationDisplay.AutoPanMode.RECENTER
+
+        // if the section has not been visited before
+        if (!sectionsVisited.containsKey(fenceFeatureName)) {
+
+            // the fence GeoElement as an ArcGISFeature
+            val fenceFeature = fenceGeotriggerNotificationInfo.fenceGeoElement as ArcGISFeature
+
+            // get the description from the feature's attributes
+            val description = fenceFeature.attributes?.getValue("description").toString()
+
+            // fetch the fence feature's attachments
+            attachmentsFuture = fenceFeature.fetchAttachmentsAsync()
+            // listen for fetch attachments to complete
+            attachmentsFuture.addDoneListener {
+                // get the feature attachments
+                val attachments = attachmentsFuture.get()
+                if (attachments.isNotEmpty()) {
+                    // get the first (and only) attachment for the feature, which is an image
+                    val imageAttachment = attachments.first()
+                    // fetch the attachment's data
+                    val attachmentDataFuture = imageAttachment.fetchDataAsync()
+                    // listen for fetch data to complete
+                    attachmentDataFuture.addDoneListener {
+                        // get the attachments data as an input stream
+                        val attachmentInputStream = attachmentDataFuture.get()
+                        // save the input stream to the device and get a reference to it's URI
+                        val attachmentImageURI =
+                            saveToInternalStorage(fenceFeatureName, attachmentInputStream)
+                        // create a new garden section data object and add it to the hash map of visited sections
+                        sectionsVisited[fenceFeatureName] =
+                            GardenSection(
+                                fenceFeature,
+                                fenceFeatureName,
+                                description,
+                                attachmentImageURI
+                            )
+                        populateUI(
+                            sectionsVisited[fenceFeatureName]!!,
```

Comment:
let's add a null check in the populateUI method and avoid using !!s if we can

Reply:
Done!